lib/protobuf/field/base_field.rb

```diff
@@ -69,10 +69,12 @@ def default
       end
 
       def default_value
-        @default_value ||= case
-                           when optional? then typed_default_value
-                           when repeated? then ::Protobuf::Field::FieldArray.new(self).freeze
-                           when required? then nil
+        @default_value ||= if optional? || required?
+                             typed_default_value
+                           elsif repeated?
+                             ::Protobuf::Field::FieldArray.new(self).freeze
+                           else
+                             fail "something went very wrong"
```

Comment:
Well this code should be unreachable actually, so it should be ominous -- it means a field isn't optional, required, or repeated. We could make change the message though to still be ominous while more descriptive :)